### PR TITLE
Fixes Make duplicated block inserts silent and not crashing due to incompat…

### LINKED ISSUE #2951: fix call to mnesia outside the transaction context in chain state
## Expected Behavior

Update chain and state in transactional way

## Actual Behavior

https://github.com/aeternity/aeternity/blob/master/apps/aecore/src/aec_chain_state.erl#L423

The check if we have a node is done outside the transaction context.
We have a case when two nodes slipped into chain state and the second one is failing internal check with found_already_calculated_state.

Result: can't sync any further

## Steps to reproduce

...on the way

### PR DESCRIPTION
**Pull request by sennui**
Fixes #2951

### CODE REVIEW

File: apps/aecore/src/aec_chain_state.erl [hunk 1]
```diff
@@ -431,7 +431,7 @@ internal_insert(Node, Block, Origin) ->
             try aec_db:ensure_transaction(Fun)
             catch exit:{aborted, {throw, ?internal_error(What)}} -> internal_error(What)
             end;
-        {ok, Node} -> ok;
+        {ok, Node} -> {ok, #{}};
```
Comment (hanssv) [inline apps/aecore/src/aec_chain_state.erl]: On line 444 there is also an incorrect return of `ok`

Comment (sennui) [reply apps/aecore/src/aec_chain_state.erl]: yes, but it's not relevant to return of the function

Comment (tolbrino) [reply apps/aecore/src/aec_chain_state.erl]: I find the semantics weird. Why not simply return an error like `already_present` or so?

**Comment by tolbrino** [comment]
@sennui Entry in release note is missing.

File: apps/aecore/src/aec_chain_state.erl [hunk 2]
```diff
@@ -431,7 +431,7 @@ internal_insert(Node, Block, Origin) ->
             try aec_db:ensure_transaction(Fun)
             catch exit:{aborted, {throw, ?internal_error(What)}} -> internal_error(What)
             end;
-        {ok, Node} -> ok;
+        {ok, Node} -> {error, already_in_db};
```
Comment (hanssv) [inline apps/aecore/src/aec_chain_state.erl]: `internal_insert_transaction` below returns `ok` in one branch - should also be `{error, already_in_db}`?

Comment (hanssv) [reply apps/aecore/src/aec_chain_state.erl]: Nevermind, I can't read.

**Pull request merged**